inference-engine/tests/unit/cpu/lru_cache.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include "cache/lru_cache.h"
+
+namespace {
+struct Key {
+    size_t hash() const {
+        return std::hash<int>().operator()(data);
+    }
+    bool operator==(const Key& rhs) const noexcept {
+        return this->data == rhs.data;
+    }
+
+    int data;
+};
+} // namespace
+
+TEST(LruCacheTests, Evict) {
+    constexpr size_t capacity = 10;
+    MKLDNNPlugin::LruCache<Key, int> cache(capacity);
+    for (size_t i = 0; i < 2 * capacity; ++i) {
+        ASSERT_NO_THROW(cache.put({10}, 10));
+    }
+    ASSERT_NO_THROW(cache.evict(5));
+    ASSERT_NO_THROW(cache.evict(10));
+    int result = cache.get({10});
+    ASSERT_EQ(result, 0);
+    ASSERT_NO_THROW(cache.evict(0));
+}
+
+TEST(LruCacheTests, Put) {
+    constexpr size_t capacity = 10;
+    MKLDNNPlugin::LruCache<Key, int> cache(capacity);
+    for (size_t i = 0; i < 2 * capacity; ++i) {
+        ASSERT_NO_THROW(cache.put({10}, 10));
+    }
+
+    ASSERT_EQ(cache.get({10}), 10);
+}
+
+TEST(LruCacheTests, Get) {
+    constexpr size_t capacity = 10;
+    MKLDNNPlugin::LruCache<Key, int> cache(capacity);
+    for (int i = 1; i < 2 * capacity; ++i) {
+        ASSERT_NO_THROW(cache.put({i}, i));
+    }
+
+    for (int i = 1; i < capacity; ++i) {
+        ASSERT_EQ(cache.get({i}), 0);
+    }
+
+    for (int i = capacity; i < 2 * capacity; ++i) {
+        ASSERT_EQ(cache.get({i}), i);
+    }
+}
+
+TEST(LruCacheTests, LruPolicy) {
```

Comment:
is it in memory cache?

Reply:
Specifically MKLDNNPlugin::LruCache just a templated dictionary of limited capacity with LRU eviction policy. This is a building block for CPU plugin runtime data cache (mainly for JIT kernels) that is developed to reduce overheads on JIT kernels generation at runtime due to shapes changing, e.g. for the case of dynamically changing shapes.

Reply:
Yep. We store cached data in RAM.